R/utilities-simulation.R

```diff
@@ -43,3 +43,38 @@ runSimulation <- function(simulation) {
   results <- clrCall(simulationRunner, "RunSimulation", simulation$ref)
   return(clrCall(results, "IndividualResultsAsArray"))
 }
+
+
+#' Adds the quantities as output into the  \code{simulation}. The quantiites can either be specified using explicit instances or using paths.
+#'
+#' @param quantities Instance of quantities to add (typically retrieved using \code{getAllQuantitiesMatching})
+#' @param paths List of quantity path to add
+#' @param simulation Instance of a simulation for which output selection should be updated.
+#'
+#' @examples
+#'
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#' outputs <- c("Organism|VenousBlood|Plasma|Caffeine", "Organism|ArterialBlood|**|Caffeine")
+#' addOutputs(paths=outputs, sim)
+#'
+#' @export
+addOutputs <- function(quantities = NULL, paths=NULL, simulation){

```

Comment:
This methods is the one that the user will be using.
I am not suire what's best from the API point of view. Having a way to add quantities instances OR paths (by having two parameters, with at least one required) or to have one parameters that can be of type string array (for path) or quantity array (for instances).

Once we have decided, I'll implement the tests

Reply:
Note: The advantaged of two parameters is that I can theoretically combined instances, and paths together. Will it be used? Probably not

Reply:
I would vote for path strings ontly... I think this is more convenient.

Reply:
why not also support instances? Let say you have retrieved a bunch of quantities using getAllParametersMatching (or getAllQuantityMatching). You have updated the start values etc.. 
Now you could just do `addOutput(quantities, sim)` and be done. No need to go back to strings. 

In this method, I had two optional inputs (quantities and paths). I am going to change the function to be quantitiesOrPaths as one parameter. That was however an example of a method that takes two optional paraemters. One of which cannot be null. Does it make sense?